include/kodlab_mjbots_sdk/limb_base.h

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file limb_base.h
+ * @author Kodlab - Zac Gong (zacgong@seas.upenn.edu)
+ * @brief Abstract class for limbs, consisting of multiple joints
+ * @version 0.1
+ * @date 2022-08-18
+ *
+ * @copyright BSD 3-Clause License, Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+ *
+ */
+
+#pragma once
+#include <limits>
+#include <memory>
+#include <vector>
+#include <type_traits>
+#include <string>
+#include "kodlab_mjbots_sdk/joint_base.h"
+#include <Eigen/Dense>
+
+namespace kodlab
+{
+    /**
+     * Abstract limb class that takes in a series of joints and treats it as an abstract limb
+     * Contains the kinematics and configurations for a leg
+     * @brief Abstract base class for legs consisting of multiple joints to be used in the robot class
+     */
+
+    struct LimbConfig
+    {
+        std::vector<std::array<float, 3>> actuator_offsets; // Offset between centers of two joints, in the frame of the first joint
+        std::vector<Eigen::Matrix3f> actuator_orientations; // Rotation matrices for each joint with respect to the previous joint (begins with R0_1)
+        Eigen::MatrixXf selection_matrix;
```

Comment:
Provide Doxygen-friendly comments, e.g., 
```suggestion
        /** 
         * @brief Offset between centers of two joints, in the frame of the 
         * first joint
         */
        std::vector<std::array<float, 3>> actuator_offsets;

        /**
         * @brief Rotation matrix transforming from the previous joint frame to
         * the current joint frame, beginning with \f${}^0 R_1\f$
         */
        std::vector<Eigen::Matrix3f> actuator_orientations;

        /** 
         * @brief Matrix of zeros and ones for re-ordering joints in member 
         * methods
         */
        Eigen::MatrixXf selection_matrix;
```

Reply:
fixed for larger formatting commit